go/vt/wrangler/materializer.go

```diff
@@ -583,46 +588,107 @@ func (wr *Wrangler) buildMaterializer(ctx context.Context, ms *vtctldatapb.Mater
 }
 
 func (mz *materializer) deploySchema(ctx context.Context) error {
+
 	return mz.forAllTargets(func(target *topo.ShardInfo) error {
+		needsCopy := false
+		copyTables := map[string]bool{}
+		targetTables := map[string]bool{}
+
 		for _, ts := range mz.ms.TableSettings {
-			tableSchema, err := mz.wr.GetSchema(ctx, target.MasterAlias, []string{ts.TargetTable}, nil, false)
+			targetTables[ts.TargetTable] = true
+
+			if ts.CreateDdl == createDDLAsCopy {
+				needsCopy = true
+				copyTables[ts.TargetTable] = true
```

Comment:
Target table and source table can be different, so the table to be copied will have to use sqlparser.TableFromStatement(ts.SourceExpression) if the source expression is present ...

Reply:
see other comment